src/planner/binder/query_node/bind_select_node.cpp

```diff
@@ -380,6 +383,20 @@ unique_ptr<BoundQueryNode> Binder::BindSelectNode(SelectNode &statement, unique_
 			auto bound_expr = group_binder.Bind(group_expressions[i], &group_type);
 			D_ASSERT(bound_expr->return_type.id() != LogicalTypeId::INVALID);
 
+			if (group_type.id() == LogicalTypeId::VARCHAR && !DBConfig::GetConfig(context).options.collation.empty()) {
+				// TODO If there is a collation on a group x, we should group by the collated expr,
+				//  but also push a first(x) aggregate)
+				info.implicit_collate[i] = result->aggregates.size();
```

Comment:
Rather than trying to figure out if a collation is present here - it would be better to check this earlier on when the collation is actually pushed into the group by (see `bind_select_node.cpp:384` where `ExpressionBinder::PushCollation` is called).

Currently this doesn't work with explicit collations on columns, e.g.:

```sql
SELECT x COLLATE NOCASE FROM t GROUP BY 1;
```

Or when the collation is part of the schema:

```sql
CREATE TABLE t(x VARCHAR COLLATE NOCASE);
INSERT INTO t VALUES ('WoRlD'), ('WORLD');
FROM t GROUP BY x;
┌─────────┐
│    x    │
│ varchar │
├─────────┤
│ world   │
└─────────┘
```